cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -714,7 +714,16 @@ func (o *ScaleUpOrchestrator) balanceScaleUps(
 	schedulablePodGroups map[string][]estimator.PodEquivalenceGroup,
 ) ([]nodegroupset.ScaleUpInfo, errors.AutoscalerError) {
 	// Recompute similar node groups in case they need to be updated
-	similarNodeGroups := o.ComputeSimilarNodeGroups(nodeGroup, nodeInfos, schedulablePodGroups, now)
+	nodeGroups := o.ComputeSimilarNodeGroups(nodeGroup, nodeInfos, schedulablePodGroups, now)
+
+	// Similar node groups may return injected node groups as they are used for simulations
+	var similarNodeGroups []cloudprovider.NodeGroup
+	for _, ng := range nodeGroups {
+		if ng.Exist() || o.scaleUpExecutor.asyncNodeGroupStateChecker.IsUpcoming(ng) {
```

Comment:
This change seems like it would be good to have a UT scenario for. Maybe in `cloudprovider/test/test_cloud_provider.go` we add a `(tcp *TestCloudProvider) AddInjectedNodeGroup` method that we can then leverage in a new test scenario in `core/scaleup/orchestrator/orchestrator_test.go`?

I'm not sure if the existing `func (tcp *TestCloudProvider) BuildNodeGroup` includes all of the necessary foo to construct an "injected node group" (we do fulfill the exists=false condition in the `AddUpcomingNodeGroup` helper method which is used in the `TestUpcomingNodesFromUpcomingNodeGroups` test case, not sure about how to get the `IsUpcoming` method to return false).

@pmendelski are you up for adding a UT case for this?